### PR TITLE
Update dependencies

### LINKED ISSUE #61: Openssl is not detetected on installation
Hello, i wont to paly around with amethyst and tried to install the recomanded tooling amethyst_tools.
unfortunately during installation i stumbled over the 'Unable to detect OpenSSL version' error.
Im using arch linux and have openssl 1.1.1 and 1.0 installed. Here is the stack backtrace:

```
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:475
   5: std::panicking::begin_panic
   6: build_script_build::main
   7: std::rt::lang_start::{{closure}}
   8: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
   9: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  10: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:392
             at libstd/rt.rs:58
  11: main
  12: __libc_start_main
  13: _start
```

Im kinda noob in such things and dont know how to proceed from here on and would appreciate any support.

### PR DESCRIPTION
**Pull request by cauebs**
There's a new version of `openssl` arriving at some linux distributions' repositories, so there's a need to update dependencies that rely on it so as to use the newer version of the bindings.

Let me know if you prefer that I don't apply `rustfmt` and I'll remove that commit.

Fixes #61

### CODE REVIEW

**Comment by Avi-D-coder** [comment]
On Arch Linux. With this patch it compiles, however `amethyst new project` spits out: 
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libcore/result.rs:945:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:475
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: rust_begin_unwind
             at libstd/panicking.rs:325
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   8: core::result::unwrap_failed
   9: std::sync::once::Once::call_once::{{closure}}
  10: std::sync::once::Once::call_inner
             at libstd/sync/once.rs:349
  11: amethyst_cli::templates::deploy
  12: amethyst_cli::new::New::execute
  13: amethyst::main
  14: std::rt::lang_start::{{closure}}
  15: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  16: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  17: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:392
             at libstd/rt.rs:58
  18: main
  19: __libc_start_main
  20: _start
```

**Comment by torkleyy** [comment]
@Avi-D-coder Thanks for checking, did it work for you before?

**Comment by torkleyy** [comment]
I don't quite understand why it panics, there is not even an unwrap in there..

**Comment by Avi-D-coder** [comment]
I only tried compiling today. Without the patch compilation fails.
```
error: failed to run custom build command for `openssl v0.9.24`
process didn't exit successfully: `/tmp/cargo-installaOTX5i/release/build/openssl-700208eb71c8ae94/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'Unable to detect OpenSSL version', /home/host/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-0.9.24/build.rs:16:14
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: failed to compile `amethyst_tools v0.7.2`, intermediate artifacts can be found at `/tmp/cargo-installaOTX5i
```

Arch is at openssl 1.1.1-1

**Comment by Avi-D-coder** [comment]
Now I am just confused. I just started from scratch and had no issues with this patch. My best guess is some how the first `cargo install` templates were not copied?

**Comment by torkleyy** [comment]
@Moxinilian we should test this one and the other PR soonish, compilation seems to fail.

File: Cargo.toml [hunk 1]
```diff
@@ -21,22 +21,22 @@ name = "amethyst"
...
-regex = "0.2.10"
-reqwest = "0.7.1"
-serde = "1.0"
-serde_derive = "1.0"
-serde_json = "1.0"
-walkdir = "2.1.4"
+ansi_term = "0.11.0"
+clap = "2.32.0"
+env_proxy = "0.2.0"
+error-chain = "0.12.0"
```
Comment (Moxinilian) [inline Cargo.toml:27]: Any specific reason to force patch versions?

**Review by Moxinilian — APPROVED** [review]
Thank you! Feels good to update dependencies.
@torkleyy If cauebs doesn't see my comment in time, I'm okay with merging immediately.
I'm also thinking about what that kind of changes means for future maintainability. I don't like that we need to manually change stuff because some distros tweaked the packages they deliver.

File: Cargo.toml (hunk 1 above)
Comment (torkleyy) [reply Cargo.toml:27]: They're not forced?

**Comment by torkleyy** [comment]
@Moxinilian I don't think this will happen often.

File: Cargo.toml (hunk 1 above)
Comment (cauebs) [reply Cargo.toml:27]: To be frank, I just used `cargo upgrade`, provided by [killercup's cargo-edit](https://github.com/killercup/cargo-edit). If you leave the versions like that, Cargo interprets them as caret requirements ([source](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio)).

**Comment by Noodlez1232** [comment]
We can also use the "vendored" feature in the Cargo.toml to allow us to compile and statically link OpenSSL.
```toml
[dependencies.openssl]
version = "0.10"
features = [ "vendored" ]
```

This could help us prevent this problem in the future. It uses the OpenSSL-src crate to do so.

**Comment by Moxinilian** [comment]
LGTM
bors r+

**Comment by Moxinilian** [comment]
Seems like a cache issue, restarting.
bors r+

**Comment by torkleyy** [comment]
Compiling statically sounds like a good idea.

**Pull request merged**